crates/ibc/src/core/ics24_host/identifier.rs

```diff
@@ -92,6 +102,11 @@ impl ChainId {
         self.revision_number
     }
 
+    /// Check if the revision number is set
+    pub fn has_revision_number(&self) -> bool {
+        self.split_chain_id().1.is_some()
+    }
```

Comment:
Instead of `has_revision_number` method (which logically should always return true), there should be e.g. `is_revision_format` method that checks if the `id` is of the specific tendermint chainId format.
It also MUST check the extractd revision number from `id` matches with the `revision_number` field, so cases like `ChainId(chainA-2, 0)` should fail.
Otherwise it gets impossible for such cases to find out if the `chainA-2` with `revision_number = 0` is valid or not?!
From what I see, ibc-go checks ChainId this way, meaning if the ChainId is in the format then 2 is definitely the `revision_number`